src/public/interfaces.ts

```diff
@@ -394,6 +403,11 @@ export interface Context {
    * Time when the user clicked on the tab
    */
   userClickTime?: number;
+
+  /**
+   * Where the user prefers the file to be opened from by default during file open
+   */
+  userFileOpenPreference?: FileOpenPreference;
```

Comment:
Where is this information stored on the client? Does it ever require making a network call or is the information always available locally?

Reply:
This is stored and read from the preferences store. It's read synchronously while passing to the context. 
No network called involved.